apps/app_rpt/rpt_config.c

```diff
@@ -1354,3 +1361,44 @@ void rpt_update_boolean(struct rpt *myrpt, char *varname, int newval)
 		rpt_event_process(myrpt);
 	}
 }
+
+int rpt_is_valid_dns_name(const char *dns_name)
+{
+	int label_length, label_start;
+
+	if (!dns_name || strlen(dns_name) > MAX_DNS_NODE_DOMAIN_LEN) {
```

Comment:
Why are we limiting the domain string length?  I don't believe there is a max in the DNS spec...